gulpfile.babel.ts

```diff
@@ -351,84 +331,107 @@ gulp.task('types', async () => {
   await exec('yarn typedefs');
 });
 
+let config: WebpackTaskOptions;
+const remoteCompatConfig = {
+  afterBuild: setupExamples,
+} as const;
+
 if (args.remote) {
-  gulp.task('sdk', () => {
-    return webpackTask({
-      devtool: 'remote',
-      disableMinification: true,
-      entry: './src/inboxsdk-js/inboxsdk-REMOTE',
-      destName: sdkFilename,
-      standalone: 'InboxSDK',
-    });
-  });
-  gulp.task(
-    'remote',
-    gulp.series('pageWorld', function impBundle() {
-      return webpackTask({
-        devtool: 'remote',
-        entry: './src/platform-implementation-js/main-INTEGRATED-PAGEWORLD',
-        destName: 'platform-implementation.js',
-        afterBuild: setupExamples,
-      });
-    })
-  );
-  gulp.task('default', gulp.series('sdk', 'remote'));
+  config = {
+    ...remoteCompatConfig,
+    devtool: false,
+    entry: {
+      [sdkFilename]: {
+        library: {
+          export: 'default',
+          name: 'InboxSDK',
+          type: OutputLibraryType.UMD,
+        },
+        import: './src/inboxsdk-js/inboxsdk-REMOTE',
+      },
+      'platform-implementation':
+        './src/platform-implementation-js/main-INTEGRATED-PAGEWORLD',
+    },
+    plugins: [
+      new webpack.SourceMapDevToolPlugin({
+        fileContext: '../',
+        // Adds hash to the end of the sourcemap URL so that
+        // remote code source maps are not cached erroneously by Sentry
+        // if and when https://github.com/getsentry/sentry/issues/54819 is patched up.
+        filename: '[file].[contenthash].map',
+        publicPath: 'https://www.inboxsdk.com/build/',
+      }),
+    ],
+  };
 } else if (args.integratedPageWorld) {
   // non-remote bundle built for compatibility with remote bundle
-  gulp.task(
-    'sdk',
-    gulp.series('pageWorld', function sdkBundle() {
-      return webpackTask({
-        devtool: 'inline-source-map',
-        entry: './src/inboxsdk-js/inboxsdk-NONREMOTE-INTEGRATED-PAGEWORLD',
-        destName: sdkFilename,
-        standalone: 'InboxSDK',
-        afterBuild: setupExamples,
-        outputLibraryType: OutputLibraryType.Var,
-      });
-    })
-  );
-  gulp.task('remote', () => {
-    throw new Error('No separate remote bundle in non-remote bundle mode');
-  });
-  gulp.task('default', gulp.parallel('sdk', 'types'));
+  config = {
+    ...remoteCompatConfig,
+    devtool: 'inline-source-map',
+    entry: {
+      [sdkFilename]: {
+        library: {
+          export: 'default',
+          name: 'InboxSDK',
+          type: OutputLibraryType.Var,
```

Comment:
We should use UMD for the integratedPageWorld to mirror the remote loaded version.

```suggestion
          type: OutputLibraryType.UMD,
```